rubix-core/src/test/java/com/qubole/rubix/core/TestCachingFileSystem.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2016. Qubole Inc
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. See accompanying LICENSE file.
+ */
+package com.qubole.rubix.core;
+
+import com.qubole.rubix.spi.CacheConfig;
+import com.qubole.rubix.spi.ClusterType;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.BufferedFSInputStream;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.RawLocalFileSystem;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+
+/**
+ * Created by sakshia on 25/11/16.
+ */
+
+public class TestCachingFileSystem
```

Comment:
Rename to `MockCachingFileSystem`. Classes with Test prefix are usually meant to be the ones carrying test cases